Swift Collections (#3030)
Codecademy/docs: content/swift/concepts/collections.md

---

> Title: 'Collections'

Author: Thanks a lot for the comments! This contribution is part of my learning of git. I tried to create a new directory for 'Collections' in my own fork but failed to do so due to permission issue. Do you know how to solve this?

Description: 'Collections in Swift are data structures that store multiple values of the same type.'
Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Collections'
  - 'Arrays'
  - 'Dictionaries'
  - 'Sets'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

**Collections** are data structures that store multiple values of the same, or different types.

Collections are can be value types or reference types, depending on the implementation.

A value type is a type whose instances are copied when they are assigned to a variable or constant, or when they are passed to a function. A reference type is a type whose instances share a single copy of the data, and use references to access it.

Swift provides three primary collection types: arrays, sets, and dictionaries.

## Syntax

> In Swift, collections are defined using different syntaxes, depending on the type of collection.

Reviewer: I would suggest leaving the descriptions as is, but having one code block before the "append, insert, remove" section with all the syntaxes documented instead of the individual inline code

* **Arrays** are ordered collections of values. They are defined using square brackets `[]`, with the values separated by commas. For example: `var fruits = [\"apple\", \"banana\", \"orange\"]`
* **Sets** are unordered collections of unique values. They are defined using the Set keyword, followed by a pair of parentheses`()`, with the values separated by commas. For example:`var colors = Set(\"red\", \"green\", \"blue\")`
* **Dictionaries** are unordered collections of key-value pairs. They are defined using square brackets `[]`, with the keys and values separated by colons`:`, and the pairs separated by commas. For example:`var scores = [\"Alice\": 90, \"Bob\": 80, \"Charlie\": 70]`

> ### `append()`,`insert()` and `remove()`

Reviewer: add an empty line after the header

Reviewer: update the other sections with the same edit

Values can be added or removed from collections with the following methods:
* `append()` adds a new value to the end of an array or a set.
* `insert()` adds a new value at a specified index in an array, or inserts a new key-value pair in a dictionary.
* `remove()` removes a value from an array or a set at a specified index, or removes a key-value pair from a dictionary by its key.

> ```

Reviewer: ```suggestion
```swift

```

// Adding values to an array
fruits.append("grape") // fruits is now ["apple", "banana", "orange", "grape"]
fruits.insert("pear", at: 1) // fruits is now ["apple", "pear", "banana", "orange", "grape"]

// Removing values from an array
fruits.remove(at: 3) // fruits is now ["apple", "pear", "banana", "grape"]
fruits.removeLast() // fruits is now ["apple", "pear", "banana"]

// Adding values to a set
colors.append("yellow") // colors is now {"red", "green", "blue", "yellow"}

// Removing values from a set
colors.remove("green") // colors is now {"red", "blue", "yellow"}

// Adding key-value pairs to a dictionary
scores["David"] = 85 // scores is now ["Alice": 90, "Bob": 80, "Charlie": 70, "David": 85]

// Removing key-value pairs from a dictionary
scores.removeValue(forKey: "Bob") // scores is now ["Alice": 90, "Charlie": 70, "David": 85]
```

## Creating a collection
In Swift, you can create a collection using literal syntax or initializer syntax.

> For example, an array of employees by the `name` of `"SpongeBob"`, `"Patrick"`, and `"Squidward"` works for “Krusty Krab” and makes a salary of 100, 50, and 200 shrimps a month respectively. Below, the `employees` variable is assigned to the value of `[Employee]`. This creates an array of instances of the class `Employee`.
> 
> The names and salaries of the employees are set through the initialization method:

Reviewer: ```suggestion
For example, in the code below, the array `employees` is declared with names and salaries through the initialization method. The `employees` variable is assigned to the value of `[Employee]`, this creates an array of instances of the class `Employee`.

```

```
var employees: [Employee] = [
    Employee(name: "SpongeBob", salary: 100),
    Employee(name: "Patrick", salary: 50),
    Employee(name: "Squidward", salary: 200)
]
```

> **Note:**:  Arrays are also named types, which are types that are given a name when defined and can be referred to as such (i.e. instances of `[Employee]` have the type `[Employee]`).

Reviewer: ```suggestion
> **Note:**:  Arrays are also named types, which are types that are given a name when defined and can be referred to as such (i.e. instances of `[Employee]` have the type `[Employee]`).

```
